particle/particle/convert.py

```diff
@@ -0,0 +1,145 @@
+    series_anti = latex_table.antiparticle
+    series_anti.index = -series_anti.index
+    return pd.concat([series_real, series_anti])
+
+def get_from_pdg_extended(filename, latexes=None):
+    'Read a file, plus a list of latex files, to produce a pandas DataFrame with particle information'
+
+    def unmap(mapping):
+        return lambda x: mapping[x.strip()]
+
+    # Convert each column from text to appropriate data type
+    PDG_converters = dict(
+        Charge=unmap(Par_mapping),
+        G=unmap(Par_mapping),
+        P=unmap(Par_mapping),
+        C=unmap(Par_mapping),
+        Anti=unmap(Inv_mapping),
+        Rank=lambda x: int(x.strip()) if x.strip() else 0,
+        ID=lambda x: int(x.strip()) if x.strip() else -1,
+        Status=unmap(Status_mapping),
+        Name=lambda x: x.strip(),
+        I=lambda x: x.strip(),  # noqa: E741
+        J=lambda x: x.strip(),
+        Quarks=lambda x: x.strip()
+    )
+
+    # Read in the table, apply the converters, add names, ignore comments
+    pdg_table = pd.read_csv(filename, comment='*', names='Mass,MassUpper,MassLower,Width,WidthUpper,WidthLower,I,G,J,P,C,Anti,'
+                            'ID,Charge,Rank,Status,Name,Quarks'.split(','),
+                            converters=PDG_converters
+                            )
+
+    # Filtering out non-particles (quarks, negative IDs)
+    pdg_table = pdg_table[pdg_table.Charge != Par.u]
+    pdg_table = pdg_table[pdg_table.ID >= 0]
+    
+    # PDG's ID should be the key to table
+    pdg_table.set_index('ID', inplace=True)
+    
+    # Note that 313 should have (892) in the name.
+    if 313 in pdg_table.index and  '(892)' not in pdg_table.loc[313, 'Name']:
+        pdg_table.loc[313, 'Name'] += '(892)'
+
+    # Some post processing to produce inverted particles
+    pdg_table_inv = pdg_table[(pdg_table.Anti == Inv.Full)
+                              | ((pdg_table.Anti == Inv.Barless)
+                                 # Maybe add?    & (pdg_table.Charge != Par.u)
+                                 & (pdg_table.Charge != Par.o))].copy()
+    pdg_table_inv.index = -pdg_table_inv.index
+    pdg_table_inv.Quarks = (pdg_table_inv.Quarks.str.swapcase()
+                            .str.replace('SQRT', 'sqrt')
+                            .str.replace('P', 'p').str.replace('Q', 'q')
+                            .str.replace('mAYBE NON', 'Maybe non')
+                            .str.replace('X', 'x').str.replace('Y', 'y'))
+    
+    # Make a combined table with + and - ID numbers
+    full = pd.concat([pdg_table, pdg_table_inv])
+
+    # Add the latex
+    if latexes is None:
+        latexes = (open_text(data, 'pdgID_to_latex.txt'),)
+
+    latex_series = pd.concat([get_from_latex(latex) for latex in latexes])
+    full = full.assign(Latex=latex_series)
+
+    # These items are not very important - can be reconstructed from the PDG
+    del full['Charge'], full['J']
+    
+    # This should be in GeV, not MeV
```

Comment:
I'm confused and the issue of units is a matter where I have very strong feelings: we should return HEP units throughout the Scikit-HEP project, and those are the standard (some do it wrong!). So MeV, nanoseconds and millimeters.

Reply:
MeV is now used, and it is included in the repr.

Reply:
Thanks!